source/1_engine/rex_engine/src/string/stringpool.cpp

```diff
@@ -0,0 +1,155 @@
+#include "rex_engine/string/stringpool.h"
+
+#include "rex_engine/diagnostics/assert.h"
+#include "rex_engine/string/stringentry.h"
+#include "rex_std/functional.h"
+#include "rex_std/internal/utility/pair.h"
+#include "rex_std/unordered_map.h"
+
+namespace rsl
+{
+  inline namespace v1
+  {
+    // https://en.wikipedia.org/wiki/Fowler–Noll–Vo_hash_function
+    rsl::hash_result hash_bytes(const void* data, rsl::count_t size) noexcept
+    {
+      const rsl::hash_result fnv_offset_basis = 2166136261u;
+      const rsl::hash_result fnv_prime        = 16777619u;
+
+      const u8* bytes = static_cast<const u8*>(data);
+
+      rsl::hash_result hash = fnv_offset_basis;
+      for(rsl::count_t i = 0; i < size; ++i)
+      {
+        hash ^= bytes[i];
+        hash *= fnv_prime;
+      }
+
+      return hash;
+    }
+
+    // Template specialization for hash a rsl::string
+    template <>
+    struct hash<rsl::string>
+    {
+      rsl::hash_result operator()(const rsl::string& s) const noexcept
+      {
+        return hash_bytes(s.data(), s.size());
+      }
+    };
+
+    // Template specialization for hash a const char*
+    template <>
+    struct hash<const char*>
+    {
+      rsl::hash_result operator()(const char* s) const noexcept
+      {
+        return hash_bytes(s, rsl::strlen(s));
+      }
+    };
+  } // namespace v1
+} // namespace rsl
+
+namespace rex
+{
+  namespace string_pool
+  {
+    using EntryMap = rsl::unordered_map<StringEntryID, StringEntry>;
+
+    //-------------------------------------------------------------------------
+    EntryMap load_entry_map()
+    {
+      EntryMap map;
+
+      for(u32 i = 0; i < static_cast<u32>(SID::MaxHardcodedNameIndex); ++i)
```

Comment:
use enum reflection for this, but ideally I'm not sure the SID enum is necessary